### PR TITLE
Support for virtual hosting with S3 and S3-like object stores

### PR DESCRIPTION
**Pull request by onkarbhat**
## Change Overview

- Get the region for a bucket using GetBucketRegionWithClient with S3ForcePathStyle set to false (which means we are using the virtual host style).

- Set S3ForcePathStyle to false when building the aws config in awsConfig().

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by onkarbhat** [comment]
This is still WIP. So I've added the hold-off-merging label. 

@vkamra @tdmanv

**Comment by Shrekster** [comment]
@onkarbhat @tdmanv @pavannd1 PTAL; this should be good for review now, verified through manual testing. Pending automated Alibaba Cloud tests in a separate PR.

Let me know if anything else needs to be done here.

File: pkg/objectstore/aws.go [hunk 1]
```diff
@@ -58,8 +58,9 @@ func awsConfig(ctx context.Context, pc ProviderConfig, s SecretAws) (*aws.Config
 	}
 	cfg = cfg.WithRegion(r)
```
Comment (julio-lopez) [inline pkg/objectstore/aws.go:59]: nit:
```suggestion
	cfg = cfg.WithRegion(r)..WithS3ForcePathStyle(false)
```

File: pkg/objectstore/aws.go [hunk 2]
```diff
@@ -58,8 +58,9 @@ func awsConfig(ctx context.Context, pc ProviderConfig, s SecretAws) (*aws.Config
...
 	if pc.Endpoint != "" {
-		cfg = cfg.WithEndpoint(pc.Endpoint).WithS3ForcePathStyle(true)
+		cfg = cfg.WithEndpoint(pc.Endpoint)
 	}
+	cfg = cfg.WithS3ForcePathStyle(false)
```
Comment (julio-lopez) [inline pkg/objectstore/aws.go:63]:
```suggestion
```

**Pull request closed without merging**